src/topology/generic_pipeline.rs

```diff
@@ -183,6 +267,12 @@ where
                     // invoke current middleware layer
                     // todo: expand support for observability or transforms
                     inspector.inspect(&items);
+                    // If any resource attributes were provided on start, set or append them to the resources
+                    if !self.resource_attributes.is_empty() {
+                        for item in &mut items {
+                            item.set_or_append_attributes(self.resource_attributes.clone())
```

Comment:
It looks like we clone the attributes every time, even if the resource may be unset. Could we pass by reference and only clone in the implementation if we need to include them?

Reply:
Yeah, that's fair :+1: